### PR TITLE
♻️ 🐛 Refactor and Bugfixes

### LINKED ISSUE #1164: 🐛After the architecture refactoring part 1 the SolutionExporer does not highlight the selected diagram
#### Describe your issue

bildschirmfoto 2018-12-05 um 08 47 16

#### How can others reproduce the issue?

- select a diagram

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [ ] I've checked if this issue already exists
- [ ] I've included all the information that i think is relevant
- [ ] I've added logs and/or screenshots (if applicable)
- [ ] I've mentioned PRs and issues that relate to this one

### LINKED ISSUE #1165: 🐛SolutionExplorer lets the User Delete Diagrams from FileSystem and ProcessEngine
#### Describe your issue

After the architecture refactoring part 1 the SolutionExporer allows the user to delete diagrams from the ProcessEngine and from the file system.

Thats not bad, but the user should get a modal which asks whether the user is sure to delete the diagram, before deleting continues or gets aborted.

#### How can others reproduce the issue?

- click on the `x` while hovering over a diagram

#### Possible solution

- add a modal which allows to continue or to abort

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [ ] I've checked if this issue already exists
- [ ] I've included all the information that i think is relevant
- [ ] I've added logs and/or screenshots (if applicable)
- [ ] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Add set and get for ActiveSolution to SolutionService and use it.
2. Fix bug when ProcessEngineRoute has changed where the navbar has shown the previous opened diagram name.
3. Fix bug when navigate to /diagram/detail from browser without a diagram name.
4. Fix bug when the heatmap gets attached without an activeDiagram.
5. Fix bug when navigate between inspect views without an active diagram.
6. Show message on heatmap If no active Diagram.
7. Remove unused EventAggregator from heatmap.
8. Fix bug which shows the inspect view on diagram detail.
9. Remove processdef-detail check on navbar.
10. Use diagram.name instead of diagram.uri for class and title binding.
11. Fix getting the wrong params from router and remove unnecessary processdef stuff.

**Issues:**

Closes #1164 
Closes #1165 

PR: #1162 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/services/SolutionService.ts [hunk 1]
```diff
@@ -25,7 +35,17 @@ export class SolutionService implements ISolutionService {
     return solutionEntry;
   }
 
-  public setActiveSolution(solution: ISolutionEntry): void {
+  public setActiveSolutionEntry(solution: ISolutionEntry): void {
     this._activeSolution = solution;
+  public getActiveDiagram(): IDiagram {
```
Comment (SteffenKn) [inline src/services/SolutionService.ts]: This looks broken

File: src/services/SolutionService.ts [hunk 2]
```diff
@@ -25,7 +35,19 @@ export class SolutionService implements ISolutionService {
...
+    return this._activeDiagram;
+  }
+
+  public setActiveDiagram(diagram: IDiagram): void {
+    this._activeDiagram = diagram;
+
+    this._eventAggregator.publish(environment.events.navBar.updateActiveSolutionAndDiagram);
+
+  }
+  }
```
Comment (SteffenKn) [inline src/services/SolutionService.ts]: This looks broken

Comment (SteffenKn) [reply src/services/SolutionService.ts]:
```suggestion
```

Comment (SteffenKn) [reply src/services/SolutionService.ts]:
```suggestion
```

File: src/modules/design/diagram-detail/diagram-detail.ts [hunk 3]
```diff
@@ -72,20 +72,16 @@ export class DiagramDetail {
...
     }
 
+    this._activeSolutionEntry = await this._solutionService.getActiveSolutionEntry();
     this.activeDiagram = await this._activeSolutionEntry.service.loadDiagram(routeParameters.diagramName);
 
-    this._eventAggregator.publish(environment.events.navBar.updateActiveSolutionAndDiagram,
-      {
-        solutionEntry: this._activeSolutionEntry,
-        diagram: this.activeDiagram,
-      });
```
Comment (SteffenKn) [inline src/modules/design/diagram-detail/diagram-detail.ts]: Remove consecutive blank line
```suggestion
```

File: src/modules/design/diagram-detail/diagram-detail.ts [hunk 4]
```diff
@@ -72,21 +72,15 @@ export class DiagramDetail {
 
   public async activate(routeParameters: RouteParameters): Promise<void> {
 
-    this._activeSolutionEntry = await this._solutionService.getActiveSolutionEntry();
-
     const diagramNameIsNotSet: boolean = routeParameters.diagramName === undefined;
     if (diagramNameIsNotSet) {
+      this._router.navigateBack();
```
Comment (SteffenKn) [inline src/modules/design/diagram-detail/diagram-detail.ts]: Add some spacing
```suggestion
      this._router.navigateBack();

```

File: src/services/SolutionService.ts [hunk 5]
```diff
@@ -25,7 +36,18 @@ export class SolutionService implements ISolutionService {
...
+
+  public getActiveDiagram(): IDiagram {
+    return this._activeDiagram;
+  }
+
+  public setActiveDiagram(diagram: IDiagram): void {
+    this._activeDiagram = diagram;
+
+    this._eventAggregator.publish(environment.events.navBar.updateActiveSolutionAndDiagram);
+
```
Comment (SteffenKn) [inline src/services/SolutionService.ts]: Remove Unneeded Spacing
```suggestion
```

File: src/modules/config-panel/config-panel.ts [hunk 6]
```diff
@@ -116,7 +120,7 @@ export class ConfigPanel {
 
     this._notificationService.showNotification(NotificationType.SUCCESS, 'Successfully saved settings!');
 
-    this._router.navigateBack();
+    this._router.navigateToRoute('start-page');
```
Comment (SteffenKn) [inline src/modules/config-panel/config-panel.ts]: Why should the BPMN-Studio navigate to the start page when saving the settings?

An example:
I am working on a local diagram and would like to deploy the diagram on a different processEngine than the connected one.
Then I save my diagram, open the settings to change the processEngine, save the settings and for some reason get redirected to the start page.
Then I would be mad that I couldn't immediately continue working on the diagram or deploy the diagram to the pe.
Instead, I would have to open the solution explorer first, search for the diagram I have been working on (if I haven't already forgotten which diagram I have been working on) and select the diagram in order to continue my work.

File: src/modules/solution-explorer/solution-explorer-solution/solution-explorer-solution.ts [hunk 7]
```diff
@@ -364,12 +365,8 @@ export class SolutionExplorerSolution {
   // TODO: This method is copied all over the place.
   public async navigateToDetailView(diagram: IDiagram): Promise<void> {
     // TODO: Remove this if cause if we again have one detail view.
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-solution/solution-explorer-solution.ts]:
```suggestion
```

File: src/modules/solution-explorer/solution-explorer-panel/solution-explorer-panel.ts [hunk 8]
```diff
@@ -286,12 +286,8 @@ export class SolutionExplorerPanel {
   // TODO: This method is copied all over the place.
   private async _navigateToDetailView(diagram: IDiagram, solution: ISolutionEntry): Promise<void> {
     // TODO: Remove this if cause if we again have one detail view.
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-panel/solution-explorer-panel.ts]:
```suggestion
```

File: src/modules/config-panel/config-panel.ts (hunk 6 above)
Comment (SteffenKn) [reply src/modules/config-panel/config-panel.ts]:
```suggestion
    this._router.navigateBack();
```

File: src/modules/design/diagram-detail/diagram-detail.ts [hunk 9]
```diff
@@ -72,21 +72,16 @@ export class DiagramDetail {
 
   public async activate(routeParameters: RouteParameters): Promise<void> {
 
-    this._activeSolutionEntry = await this._solutionService.getActiveSolutionEntry();
-
     const diagramNameIsNotSet: boolean = routeParameters.diagramName === undefined;
     if (diagramNameIsNotSet) {
+      this._router.navigateToRoute('start-page');
```
Comment (SteffenKn) [inline src/modules/design/diagram-detail/diagram-detail.ts]: Missing newline:

```suggestion
      this._router.navigateToRoute('start-page');

```

File: src/modules/config-panel/config-panel.ts [hunk 10]
```diff
@@ -92,6 +95,7 @@ export class ConfigPanel {
     }
 
     this._eventAggregator.publish(environment.events.configPanel.processEngineRouteChanged, this.baseRoute);
+    this._solutionService.setActiveDiagram(undefined);
```
Comment (cmg-dev) [inline src/modules/config-panel/config-panel.ts]: Please leave a comment why we want to do this.

**Pull request merged**